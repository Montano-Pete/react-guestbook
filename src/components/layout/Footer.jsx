const Footer = () => {
  return (
    <footer>
      <p>Jank Guestbook made by yours truly</p>
    </footer>
  );
};
