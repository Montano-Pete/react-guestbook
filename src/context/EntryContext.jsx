import { createContext, useState } from 'react';

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
  const [entry, setEntry] = useState();

  return (
    <EntryContext.Provider value={{ entry, setEntry }}>
      {children}
    </EntryContext.Provider>
  );
};

export { EntryProvider, EntryContext };
