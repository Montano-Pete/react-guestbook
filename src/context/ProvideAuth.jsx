import React from 'react';
import { createContext, useState } from 'react';

const authContext = createContext();

function ProvideAuth({ children }) {
  const [user, setUser] = useState(null);

  const login = (email, password) => {
    const loginSuccessful =
      email === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser({ email });
    return loginSuccessful;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  return (
    <authContext.Provider value={{ user, login, logout }}>
      {children}
    </authContext.Provider>
  );
}

export { authContext, ProvideAuth };
