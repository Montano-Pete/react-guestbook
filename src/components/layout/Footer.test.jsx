import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Footer from './Footer';
import { UserProvider } from '../../context/UserContext';

it('should render footer component', async () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <Footer />
      </UserProvider>
    </MemoryRouter>
  );

  const footer = await screen.findByText('Jank Guestbook made by yours truly');

  expect(footer).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
