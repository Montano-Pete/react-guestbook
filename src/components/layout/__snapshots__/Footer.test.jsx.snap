// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render footer component 1`] = `
<div>
  <footer>
    <p>
      Jank Guestbook made by yours truly
    </p>
  </footer>
</div>
`;
